### DISCUSSION TITLE
Multiple credentials

### DISCUSSION THREAD

**Discussion by robertnicjoo** [imdhemy/laravel-in-app-purchases#369] [Q&A] [answered]
I have multiple app that using same back-end. Is there a way to set different credential path for different requests?

```php
$response = Product::googlePlay()
->packageName('com.example.abc')
// is there an option available here to change credential path value???
->id($productId)
->token($token)
->get();
```

**Comment by robertnicjoo** [comment]
Over a year and no response!

**Comment by imdhemy** [comment]
Here's how it works with multiple apps. The JSON key file needs to be decoded into an array.

```php
// create a client with the key decoded into an array
$client = ClientFactory::createWithJsonKey(jsonKey: $jsonKey, scopes: $scopes);

// pass the $client to the facade
Subscription::googlePlay($client)->id($itemId)->token($purchaseToken)->get();
```
I hope this helps!

**Reply by robertnicjoo** [reply]
> ```
> jsonKey: $jsonKey, scopes: $scopes
> ```

@imdhemy  Thanks for the response, I know that `$jsonKey` is location my JSON file is but:

1. Do i really need `scopes`?
2. If yes, can you give an example of it?

**Reply by imdhemy** [reply, marked as answer]
No, the default scopes are enough for In-App purchase APIs. 

https://github.com/imdhemy/google-play-billing/blob/8fe22538c7a117a46e9a959c3819de13dcbae536/tests/ClientFactoryTest.php#L40-L42
